nesta/packages/crunchbase/crunchbase_collect.py

```diff
@@ -0,0 +1,333 @@
+from contextlib import contextmanager
+import logging
+import pandas as pd
+import re
+import requests
+import tarfile
+from tempfile import NamedTemporaryFile
+
+from nesta.production.luigihacks import misctools
+from nesta.production.orms.orm_utils import insert_data
+from nesta.packages.geo_utils.country_iso_code import country_iso_code_to_name
+from nesta.packages.geo_utils.geocode import generate_composite_key
+
+
+@contextmanager
+def crunchbase_tar():
+    """Downloads the tar archive of Crunchbase data.
+
+    Returns:
+        :code:`tarfile.Tarfile`: opened tar archive
+    """
+    crunchbase_config = misctools.get_config('crunchbase.config', 'crunchbase')
+    user_key = crunchbase_config['user_key']
+    url = 'https://api.crunchbase.com/v3.1/csv_export/csv_export.tar.gz?user_key='
+    with NamedTemporaryFile() as tmp_file:
+        r = requests.get(''.join([url, user_key]))
+        tmp_file.write(r.content)
+        try:
+            tmp_tar = tarfile.open(tmp_file.name)
+            yield tmp_tar
+        finally:
+            tmp_tar.close()
+
+
+def get_csv_list():
```

Comment:
Depends how much you want to... but it might be a good idea to refactor some of the guts of this function this into a shared package and call it `list_tarfile_contents(tarfile, regex_pattern=r'^(.*)$')`. Not sure how much it would get used, conceivably it could though.

Reply:
I am going to leave this for now